pkg/registry/generic/rest/proxy.go

```diff
@@ -202,7 +206,51 @@ func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 
 		return tlsConn, nil
 	default:
-		return nil, fmt.Errorf("Unknown scheme: %s", h.Location.Scheme)
+		return nil, fmt.Errorf("unknown scheme: %s", h.Location.Scheme)
+	}
+}
+
+func dialURLWithDialer(httpTransport *http.Transport, location *url.URL) (net.Conn, error) {
```

Comment:
> I still think ease of reading is worth the extra lines

My concern is people fixing something in one place and not the other. If you think it's tested well enough, then it's fine.